Source/Libraries/FaultData/DataAnalysis/DataSeries.cs

```diff
@@ -442,6 +443,40 @@ public void Downsample(int maxSampleCount)
             DataPoints = data;
         }
 
+        /// <summary>
+        /// Upsamples the current DataSeries to requested sample count, assuming the requested rate is larger than the current
+        /// </summary>
+        /// <param name="maxSampleCount"></param>
+        public void Upsample(int sampleCount)
+        {
+            // don't actually upsample, if it doesn't need it.
+            if (sampleCount <= DataPoints.Count()) return;
+
+            List<double> xValues = DataPoints
+                .Select(point => (double) point.Time.Subtract(StartTime).Ticks)
+                .ToList();
+
+            List<double> yValues= DataPoints
+                .Select(point => point.Value)
+                .ToList();
+
+            SplineFit splineFit = SplineFit.ComputeCubicSplines(xValues, yValues);
+
+            long tickIncrement = EndTime.Subtract(StartTime).Ticks / sampleCount;
+
+            List<DataPoint> data = Enumerable
+                .Range(0, sampleCount)
+                .Select(sample =>
+                    new DataPoint() 
+                    { 
+                        Time = StartTime.AddTicks(tickIncrement * sample),
+                        Value = splineFit.CalculateY(tickIncrement * sample) 
+                    }
+                ).ToList();
```

Comment:
The cumulative error in this computation adds up pretty quick. Upsampling to 16 samples per cycle over 10 cycles builds up more than 100 microseconds of error over the course of that time window. We shouldn't be at any risk of overflow so there shouldn't be any issue doing the multiplication before the division in the main loop.

```suggestion
            long durationTicks = EndTime.Subtract(StartTime).Ticks;

            List<DataPoint> data = Enumerable
                .Range(0, sampleCount)
                .Select(sample => sample * durationTicks / sampleCount)
                .Select(sampleTicks =>
                    new DataPoint() 
                    { 
                        Time = StartTime.AddTicks(sampleTicks),
                        Value = splineFit.CalculateY(sampleTicks) 
                    }
                ).ToList();
```